pkg/utils/nodeclaim/nodeclaim.go

```diff
@@ -232,3 +205,45 @@ func UpdateNodeOwnerReferences(nodeClaim *v1beta1.NodeClaim, node *v1.Node) *v1.
 	})
 	return node
 }
+
+// TerminationHandler is a helper function that takes a v1beta1.NodeClaim and calls cloudProvider.Delete() if status condition
+// on nodeClaim is not terminating. If it is terminating then it will check if the kubelet process is still running. It will
+// return an error and a boolean that indicates if the reconciliation should be re-enqueued.
+func TerminationHandler(ctx context.Context, c client.Client, nodeClaim *v1beta1.NodeClaim, node *v1.Node, cloudProvider cloudprovider.CloudProvider) (bool, error) {
+	// Check if the status condition on nodeClaim is Terminating
+	terminationStatus := nodeClaim.StatusConditions().GetCondition(v1beta1.Terminating)
+	if terminationStatus.IsUnknown() || terminationStatus.IsFalse() {
+		// If not then call Delete on cloudProvider to trigger termination
+		if err := cloudProvider.Delete(ctx, nodeClaim); cloudprovider.IgnoreNodeClaimNotFoundError(err) != nil {
+			return false, fmt.Errorf("terminating cloudprovider instance, %w", err)
+		}
+		if err := SetTerminatingStatus(ctx, c, nodeClaim); err != nil {
+			return false, err
+		}
+		return true, nil
```

Comment:
```suggestion
		return true, SetTerminatingStatus(ctx, c, nodeClaim)
```

Why do we not want to requeued if there is an error?

Reply:
Fixed as per our discussion